weatherman/weatherman.js

```diff
@@ -0,0 +1,134 @@
+import { getMaxObj, getMinObj, getAvgObj } from "./utils/mathUtils.js"
+import { getYear, formattedDateForOutput, getShortMonthFromString } from "./utils/dateUtils.js"
+import { FilReader } from "./fileReader.js"
+import { YEAR_WEATHER_STATS_COLUMNS, MONTH_WEATHER_STATS_COLUMNS} from "./constants.js";
+
+/**
+ * WeatherMan class .
+ */
+export class WeatherMan {
+
+    /**
+     * Function Constructor.
+     * @param { String } directoryPath -Path where files are stored.
+     * @param { String } date -Date in form of year or Y/m.
+     */
+    constructor(directoryPath, date) {
+        this.directoryPath = directoryPath
+        this.date = date
+    } 
+
+     /**
+     * Parse date string 
+     */
+    parseDate() {
+        this.formatToMatchFiles = getYear(this.date)
+        if (this.withMonth) {
```

Comment:
is `withMonth` an attribute of WeatherMan class?

Reply:
flag set in calculateMonthStats function